compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPULowerExecutableTarget.cpp

```diff
@@ -86,8 +86,25 @@ getPipelineOptions(FunctionOpInterface funcOp,
   if (DictionaryAttr config = translationInfo.getConfiguration()) {
     if (config.contains(LLVMGPUAttrNames::kNoReduceSharedMemoryBankConflicts))
       pipelineOptions.enableReduceSharedMemoryBankConflicts = false;
-    if (config.contains(LLVMGPUAttrNames::kNoReorderWorkgroups))
-      pipelineOptions.enableReorderWorkgroups = false;
+    if (config.contains(LLVMGPUAttrNames::kReorderWorkgroups)) {
+      // Get the workgroups reorder config and enable the workgroup reordering
+      Attribute reorderGroupOption =
+          config.get(LLVMGPUAttrNames::kReorderWorkgroups);
+      assert(isa<StringAttr>(reorderGroupOption) &&
+             "reorder strategy should be a StringAttr");
+      StringRef reorderStr =
+          llvm::cast<StringAttr>(reorderGroupOption).getValue();
+      if (reorderStr == "transpose") {
+        pipelineOptions.reorderStrategy = ReorderWorkgrupsStrategy::Transpose;
+      } else if (reorderStr == "swizzle") {
+        pipelineOptions.reorderStrategy = ReorderWorkgrupsStrategy::Swizzle;
+      } else {
+        if (reorderStr != "none")
+          funcOp.emitOpError("Unhandled reorder option");
```

Comment:
I don't remember the exact syntax, but this error can be made more helpful and precise:
```suggestion
          funcOp.emitOpError() << "Unknown " << LLVMGPUAttrNames::kReorderWorkgroups << "value: " << reorderGroupOpton;
```